### ISSUE TITLE
Support for sass modules

### ISSUE THREAD

**Issue by stof** [scssphp/scssphp#55]
Modules are the new way of organizing sass files, current available in dart-sass. See http://sass.logdown.com/posts/7858341-the-module-system-is-launched for the announcement.

Is there any plan for implementing them in scssphp ?

**Comment by robocoder** [comment]
By "plan", do you mean something more concrete than a wish/todo list?  ;)

`@use` looks non-trivial though.

**Comment by stof** [comment]
it is indeed non-trivial. And currently, libsass does not have support for it yet either (only the reference implementation dart-sass has it for now).
But I expect that libraries will start looking into it (there is already some discussion about it on material-components-web for instance).

Non-support in libsass might still be considered as a blocker for now by many projects, due to node-sass (the libsass binding) being used a lot more than sass (the dart-sass JS distribution) for projects managing their tooling via npm (see https://www.npmtrends.com/node-sass-vs-sass). But the trend seems to be in favor of dart-sass (node-sass having lots of existing projects which haven't switched their implementation just because dart-sass is released).

**Comment by Cerdic** [comment]
Definitely a *huge* re-work of ScssPHP will be needed. 
Time will say if someone is able to get enough available time to go on it.

**Comment by phuclh** [comment]
Any updates on this? Thank you!

**Comment by stof** [comment]
@phuclh this won't happen in 1.x. The code of scssphp is too messy (and too far from being spec compliant) to add modules in it.

The work on 2.0 is in progress, rewriting the compiler based on the dart-sass code, bringing us to spec compliance. Once this is done (no ETA as I'm doing that on my free time), supporting modules will be the next big work (so probably coming in 2.1.0 or 2.2.0).
See https://github.com/scssphp/scssphp/milestone/2?closed=1 for the work already done on it.

**Comment by phuclh** [comment]
> @phuclh this won't happen in 1.x. The code of scssphp is too messy (and too far from being spec compliant) to add modules in it.
> 
> The work on 2.0 is in progress, rewriting the compiler based on the dart-sass code, bringing us to spec compliance. Once this is done (no ETA as I'm doing that on my free time), supporting modules will be the next big work (so probably coming in 2.1.0 or 2.2.0).
> See https://github.com/scssphp/scssphp/milestone/2?closed=1 for the work already done on it.

Thank you for quick response and good to know you are working on the version 2 😊.

**Comment by wtfred** [comment]
Hello Stof,

I would like to know, if there is still no ETA for a 2.0? I'm looking forward for `@use` and `@forward` 
Thanks a lot!

**Comment by wtfred** [comment]
Ok, thanks for your answer and your work! :+1:

**Comment by neoacevedo** [comment]
There is a workaround? idk, an '@import' or similar?

**Comment by stof** [comment]
using `@import` is supported in scssphp (`@import` is not about Sass modules, it is the old way of using multiple files in Sass)
